python/cudf/cudf/core/index.py

```diff
@@ -696,6 +696,9 @@ def copy(self, deep=True):
     def __sizeof__(self):
         return self._values.__sizeof__()
 
+    def _memory_usage(self, deep=False):
+        return self._values._memory_usage(deep=deep)
```

Comment:
You need to do this for more than just the GenericIndex. And should special case RangeIndex to return 0.

Reply:
Sorry about that - I didn't realize I was targeting Generic rather than the base Index class.  Hopefully this is corrected now.  I also added support for categoricals (just summing the categories and codes).

> And should special case RangeIndex to return 0.

The value returned by RangeIndex doesn't seem very important.  However, pandas seems to return "`80`".  Are we okay with returning something different?

Reply:
Yes, because the RangeIndex isn't using GPU memory, whereas it is using CPU memory which we're not tracking currently.